float32 degree
float32 timeout
---
bool success